compiler/include/ForLoop.h

```diff
@@ -132,6 +135,19 @@ class ForLoop final : public LoopStmt
   CallExpr*              blockInfoGet()                      const override;
   CallExpr*              blockInfoSet(CallExpr* expr)              override;
 
+  AList&                 shadowVariables() override;
+
+  bool needToHandleOuterVars() const override { return true; }
+  BlockStmt* loopBody() const override {
+    return const_cast<ForLoop*>(this);
+  }
+  bool needsInitialAccumulate() const override { return false; }
+  Expr* asExpr() override { return this; }
+  bool isInductionVar(Symbol* sym) override;
+
+  virtual bool isForallStmt() override { return false; }
+  virtual ForallStmt *forallStmt() override { INT_ASSERT(false); return nullptr; }
```

Comment:
Nit, but for these two, consider:

- `final`
- dropping `virtual` (https://stackoverflow.com/questions/39932391/should-i-use-virtual-override-or-both-keywords)
- Not asserting for `forallStmt` and just returning `nullptr`.